homeassistant/components/wilight/switch.py

```diff
@@ -92,21 +93,24 @@ async def async_setup_entry(
     async def set_watering_time(entity, service: Any) -> None:
         if not isinstance(entity, WiLightValveSwitch):
             raise ValueError("Entity is not a WiLight valve switch")
-        watering_time = service.data[ATTR_WATERING_TIME]
-        await entity.async_set_watering_time(watering_time=watering_time)
+        if (entity.entity_id in service.data[ATTR_ENTITY_ID]):
```

Comment:
Why did you add this check? This check won't do anything as it will always be true. The entity service helper looks up the entity with the entity_id before we enter this function.

My previous comment was regarding the tests. Please add a test that calls a service with an entity_id targeting an entity  that is not supported by that service. Eg, calling set_watering_time on a WiLightValvePauseSwitch entity.

Reply:
Sorry for the misunderstanding.
Removed the check.
Included the test in test_switch.py, but the raise makes the test fail. Please, am I doing something wrong?

Please would you have a more elegant solution than "raise ValueError("Entity is not a WiLight valve switch")". 
I'm thinking of leaving the log less loaded than what happens with the raise.
Is there any function that displays information on the front end?
Assuming there is a function for this (Show_in_front_end)...
Something like:
 _LOGGER.warning("Entity %s is not a WiLight valve switch", entity.entity_id)
Show_in_front_end("Entity %s is not a WiLight valve switch", entity.entity_id)

Thanks.

Reply:
I was doing something basic wrong.
Now the test is OK.